.github/workflows/bench-localhost.yml

```diff
@@ -0,0 +1,104 @@
+name: "Bench HTC Mock on localhost"
+
+on:
+  push:
+    branches:
+      - "ts/add-bench-workflow"
+      - "main"
+  workflow_dispatch:
+    inputs:
+      ntasks:
+        description: "Number of tasks to be created"
+        default: "100"
+        required: false
+      subtasks-levels:
+        description: "Levels of subtasking"
+        default: "1"
+        required: false
+      purge-data:
+        description: "Whether to purge session"
+        default: "false"
+        required: false
+  release:
+    types: [created, published]
+
+jobs:
+  bench:
+    name: Get Bench session stats
+    runs-on: ubuntu-latest
+    outputs:
+      terraform-output: ${{ steps.deploy.outputs.terraform-output }}
+      armonik-endpoint: ${{ steps.get-armonik-endpoint.outputs.endpoint }}
+    steps:
+        
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+
+      - name: Install Dependencies
+        uses: aneoconsulting/ArmoniK.Action.Deploy/dependencies@main
+        with:
+          terraform: true
+          k3s: true
+          docker: true
+
+      - name: Define runner env variables
+        run: |
+          echo "core-version=$(cat versions.tfvars.json | jq '.armonik_versions.core' | sed -r 's/"//g')" >> $GITHUB_ENV
+          echo "ntasks=${{ inputs.ntasks || 500 }}" >> $GITHUB_ENV
+          echo "purge-data=${{ inputs.purge-data || false }}" >> $GITHUB_ENV
```

Comment:
I am not sure || works, but even if it does, you should take the default value defined in the parameter itself.

Reply:
Workflows do not consider inputs if they're triggered by an event other than workflow_dispatch or worfklow_call